File: flutter/lib/models/panne.dart
import 'tpe.dart';
import 'type_panne.dart';
import 'station.dart';

class Panne {
  final int id;
  final Tpe tpe;
  final TypePanne typePanne;
  final Station? station;
  final String? description;
  final String statut;
  final String dateDeclaration;

  Panne({
    required this.id,
    required this.tpe,
    required this.typePanne,
    this.station,
    this.description,
    required this.statut,
    required this.dateDeclaration,
  });

  factory Panne.fromJson(Map<String, dynamic> json) {
    return Panne(
      id: json['id'],
      tpe: Tpe.fromJson(json['tpe']),
      typePanne: TypePanne.fromJson(json['typePanne']),
      station: json['station'] != null ? Station.fromJson(json['station']) : null,
      description: json['description'],
      statut: json['statut'] ?? 'DECLAREE',
      dateDeclaration: json['dateDeclaration'] ?? '',
    );
  }
}

File: backend/src/main/java/model/Panne.java
package com.chams.tpe_backend.model;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "panne")
public class Panne {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "tpe_id", nullable = false)
    private Tpe tpe;

    @ManyToOne
    @JoinColumn(name = "type_panne_id", nullable = false)
    private TypePanne typePanne;

    @ManyToOne
    @JoinColumn(name = "station_id", nullable = true)
    private Station station;

    private String description;

    @Column(nullable = false)
    private LocalDateTime datedeclaration = LocalDateTime.now();

    @Enumerated(EnumType.STRING)
    private StatutPanne statut = StatutPanne.DECLAREE;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Tpe getTpe() { return tpe; }
    public void setTpe(Tpe tpe) { this.tpe = tpe; }
    public TypePanne getTypePanne() { return typePanne; }
    public void setTypePanne(TypePanne typePanne) { this.typePanne = typePanne; }
    public Station getStation() { return station; }
    public void setStation(Station station) { this.station = station; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    public LocalDateTime getDateDeclaration() { return datedeclaration; }
    public void setDateDeclaration(LocalDateTime dateDeclaration) { this.datedeclaration = dateDeclaration; }
    public StatutPanne getStatut() { return statut; }
    public void setStatut(StatutPanne statut) { this.statut = statut; }
}